File: pocketr-java.api/src/main/java/com/decrux/pocketr/api/services/account/OpeningBalanceService.java
package com.decrux.pocketr.api.services.account;

import com.decrux.pocketr.api.entities.db.auth.User;
import com.decrux.pocketr.api.entities.db.ledger.Account;
import java.time.LocalDate;

public interface OpeningBalanceService {

    void createForNewAssetAccount(
        User owner,
        Account assetAccount,
        long openingBalanceMinor,
        LocalDate txnDate
    );
}

File: pocketr-api/src/main/kotlin/com/decrux/pocketr/api/services/account/OpeningBalanceService.kt
package com.decrux.pocketr.api.services.account

import com.decrux.pocketr.api.entities.db.auth.User
import com.decrux.pocketr.api.entities.db.ledger.Account
import java.time.LocalDate

interface OpeningBalanceService {
    fun createForNewAccount(
        owner: User,
        account: Account,
        openingBalanceMinor: Long,
        txnDate: LocalDate,
    )
}
